### PR TITLE
[coinbase] fetchLedger

### PR DESCRIPTION
**Pull request by npomfret**
@kroitor pls squash.

Still getting to grips with the ledger format.  Please let me know what I've not done right...

### CODE REVIEW

**Comment by frosty00** [comment]
if you could change this line:

```
this.commonCurrencyCode (currencyId);
```
->
```
this.safeCurrencyCode (currencyId, currency)
```

that would be appreciated.

Currently in the process of moving the whole library to the new method.

File: js/coinbase.js [hunk 1]
```diff
@@ -637,6 +638,360 @@ module.exports = class coinbase extends Exchange {
         return this.parseBalance (result);
     }
 
+    async fetchLedger (code = undefined, since = undefined, limit = undefined, params = {}) {
+        const accountId = this.safeString2 (params, 'account_id', 'accountId');
```
Comment (kroitor) [inline js/coinbase.js]: We should rework this as shown here:

- https://github.com/ccxt/ccxt/pull/5436/files#diff-3624f6ace8416a4a6ae340743d6421c5R218

Or here

- https://github.com/ccxt/ccxt/blob/master/js/kucoin.js#L1338

**Comment by npomfret** [comment]
The `this.commonCurrencyCode` change is done.

But the `accountId` change is not.  I decided for the time being it was better to be consistent with the rest of the class.  Making the change everywhere within this class should probably be a PR on its own.

**Comment by kroitor** [comment]
> But the accountId change is not. I decided for the time being it was better to be consistent with the rest of the class. Making the change everywhere within this class should probably be a PR on its own.

The problem is it's hard to merge code like this, because people start copying it all around. I hope to find the time to rework this asap.

**Comment by npomfret** [comment]
I have more work to do on Coinbase so will add it shortly.

**Comment by npomfret** [comment]
@kroitor pls review again - I added a change to remove some of the duplication.

File: js/coinbase.js [hunk 2]
```diff
@@ -279,54 +288,37 @@ module.exports = class coinbase extends Exchange {
 
     async fetchMySells (symbol = undefined, since = undefined, limit = undefined, params = {}) {
         // they don't have an endpoint for all historical trades
-        const accountId = this.safeString2 (params, 'account_id', 'accountId');
-        if (accountId === undefined) {
-            throw new ArgumentsRequired (this.id + ' fetchMyTrades requires an account_id or accountId extra parameter, use fetchAccounts or loadAccounts to get ids of all your accounts.');
-        }
+        const request = await this.prepareAccountRequest (symbol, limit, params);
```
Comment (kroitor) [inline js/coinbase.js]: Unfortunately, we can't use `prepareAccountRequest` interchangeably for both symbols and currency codes, so, in this case, we might need to revert it.

Comment (kroitor) [reply js/coinbase.js]: The same is true for `fetchMyBuys` below.

Comment (npomfret) [reply js/coinbase.js]: Of course.  Thanks

Comment (npomfret) [reply js/coinbase.js]: Have made an amendment

**Comment by npomfret** [comment]
@kroitor any chance of getting this in?

**Comment by kroitor** [comment]
@npomfret yep, i'll test on my side, will hopefully merge today.

**Comment by npomfret** [comment]
Sorry for nagging.  But I'm keen to get this in as I've got some more ledgers to tackle.  Want to get one correct first.

**Comment by kroitor** [comment]
@npomfret pardon again for the delay, will merge it first thing within a few hours. Thx for your patience!

**Pull request merged**